packages/xchain-client/src/types.ts

```diff
@@ -57,6 +58,13 @@ export type TxParams = {
   memo?: string // optional memo to pass
 }
 
+export type FeeOptions = {
+  feeOptionKey?: FeeOptionKey
+  feeRate?: number
+  gasPrice?: BaseAmount
+  gasLimit?: BigNumber
```

Comment:
It's not generic, since `gasLimit` depends on `ethers.BignNumber`. Or in other words: `FeeOptions` includes a value, which is needed for `xchain-ethereum` only.

Reply:
To workaround it: Avoid import of `ethers` by using `bignumber.js` (we already use it in other `xchain-*` packages)

BERFORE
```ts
import BigNumber from 'bignumber.js'
```
CHANGE
```ts
import { BigNumber } from 'ethers'
```

Reply:
true makes sense.

Reply:
It seems current implementation of `transfer` in `xchain-ethereum` needs to be updated as well to use a correct `BigNumber`.

Pls check if we can merge `TxParams & FeeOptions` to have one type only: `TxParam`

Reply:
We can, and I did in #376

Reply:
okey cool, lets put that PR on hold for now then. lets wait for your refactor PR